File: python/leetcode/find-the-difference.py
class Solution:
    def findTheDifference(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: str
        """
        seen1 = {}
        seen2 = {}
        for c in s:
            seen1[c] = seen1[c] + 1 if c in seen1 else 1
        for c in t:
            seen2[c] = seen2[c] + 1 if c in seen2 else 1

        for key in seen2:
            if key not in seen1 or seen2[key] != seen1[key]:
                return key
        return -1

File: javascript/leetcode/find-the-difference.js
/**
 * @param {string} s
 * @param {string} t
 * @return {character}
 */
var findTheDifference = function(s, t) {
    seen1 = {}
    seen2 = {}
    for (var i = 0; i < s.length; i++) {
        seen1[s[i]] = seen1[s[i]] ? seen1[s[i]] + 1 : 1
    }
    for (var i = 0; i < t.length; i++) {
        seen2[t[i]] = seen2[t[i]] ? seen2[t[i]] + 1 : 1
    }

    for (key in seen2) {
        if (seen2[key] !== seen1[key]) {
            return key
        }
    }
    return -1
};